invenio_drafts_resources/records/api.py

```diff
@@ -32,39 +41,52 @@ class Record(RecordBase):
     # Configuration
     model_cls = None
 
-    pid = PIDField('id', provider=DraftRecordIdProviderV2)
-
-    conceptpid = PIDField('conceptid', provider=DraftRecordIdProviderV2)
+    pid = PIDField('id', provider=DraftRecordIdProviderV2, delete=True)
 
     is_published = PIDStatusCheckField(status=PIDStatus.REGISTERED)
 
+    #: List of fields to copy from the draft on create.
+    create_from_draft = []
+
+    #: List of fields to copy from the draft on update.
+    update_from_draft = []
+
+    # TODO: Below three methods create_from(), update_from() and register() has
+    # to be refactored. They are accessing "parent" but it may not be defined.
+    # Instead, this work should be delegated to the system fields, but it's not
+    # easy to add a new pre/post_create_from pre/post_register hook.
     @classmethod
-    def create_or_update_from(cls, draft):
-        """Create of update the record based on the draft content."""
-        try:
-            # New version
-            record = cls.get_record(draft.id)
-        except NoResultFound:
-            # New revision
-            record = cls.create(
-                {}, id_=draft.id, pid=draft.pid, conceptpid=draft.conceptpid)
-
-        # NOTE: Merge pid/conceptpid into the current db session if not already
-        # in the session.
+    def create_from(cls, draft):
```

Comment:
Not pretty - it would be better to have something on the system fields, but the hooks are not easily extended with new ones in Invenio-Records.